server/base/src/main/java/org/apache/accumulo/server/metadata/ConditionalTabletsMutatorImpl.java

```diff
@@ -78,6 +83,13 @@ public ConditionalTabletsMutatorImpl(ServerContext context,
       Function<DataLevel,String> tableMapper) {
     this.context = context;
     this.tableMapper = Objects.requireNonNull(tableMapper);
+    this.sharedWriter = Suppliers.memoize(() -> {
```

Comment:
Seems like the creation of this sharedWriter is not in the correct place, probably needs to be in ServerAmpleImpl.  For code like [this](https://github.com/apache/accumulo/blob/7ca25836e11a46fd312baa11fc4c191f1369bb87/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/split/UpdateTablets.java#L202) when it calls `getAmple().conditionallyMutateTablets()` that would still create a new ConditionalWriter.  
Maybe something like `Map<DataLevel, Supplier<ConditionalWriter>>` needs to be passed into this constructor from ServerAmpleImpl.

Reply:
Okay I think i fixed things in 9de3982. I didn't use a map since it would just be the two entries but could easily change things if you think that would be better.